### PR TITLE
[SYCLomatic] Refine the condition of removing using declaration

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -13529,38 +13529,44 @@ void NamespaceRule::runRule(const MatchFinder::MatchResult &Result) {
...
         End, SM, DpctGlobalInfo::getContext().getLangOpts());
     Len = SM.getFileOffset(End) - SM.getFileOffset(Beg) + Toklen;
-    auto Iter = MapNames::MathFuncNameMap.find(UD->getNameAsString());
-    if (Iter != MapNames::MathFuncNameMap.end()) {
-      DpctGlobalInfo::getInstance().insertHeader(UD->getBeginLoc(), HT_Math);
-      std::string Repl{"using "};
-      Repl += Iter->second;
+
+    std::unordered_set<const clang::UsingShadowDecl *> IntroducedUSD;
+    for (const auto &child : UD->getDeclContext()->decls()) {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ASTTraversal.cpp:13534]: Please try to merge the two for(){} to make code simpler.

Comment (zhiweij1) [reply clang/lib/DPCT/ASTTraversal.cpp:13534]: Done

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**